yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,209 @@
+    clean_html,
+    int_or_none,
+    join_nonempty,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class TapTapBaseIE(InfoExtractor):
+    _X_UA = 'V=1&PN=WebApp&LANG=zh_CN&VN_CODE=102&LOC=CN&PLT=PC&DS=Android&UID={uuid}&OS=Windows&OSV=10&DT=PC'
+    _VIDEO_API = 'https://www.taptap.cn/webapiv2/video-resource/v1/multi-get'
+    _INFO_API = None
+    _INFO_QUERY_KEY = 'id'
+    _DATA_PATH = ('data')
+    _ID_PATH = None
+    _META_PATH = None
+
+    def _get_api(self, url, video_id, query, **kwargs):
+        rand_hex = lambda digits: ''.join(f'{random.randint(0, 15):x}' for _ in range(digits))
+        uuid = '-'.join(rand_hex(digits) for digits in [8, 4, 4, 4, 12])
+        query = {**query, 'X-UA': self._X_UA.format(uuid=uuid)}
+        return self._download_json(url, video_id, query=query, **kwargs)
+
+    def _extract_video(self, video_id, is_intl=False):
+        video_data = self._get_api(self._VIDEO_API, video_id, query={'video_ids': video_id})['data']['list'][0]
+
+        video_url = traverse_obj(video_data, ('play_url', ('url_h265', 'url'), {url_or_none}))[0]
+        formats = self._extract_m3u8_formats(video_url, video_id)
+        for format in formats:
+            if re.search(r'^(hev|hvc|hvt)\d', format.get('vcodec', '')):
+                format['format_id'] = join_nonempty(format.get('format_id'), 'h265', delim='_')
+
+        return {
+            'id': str(video_id),
+            'formats': formats,
+            **traverse_obj(video_data, ({
+                'duration': ('info', 'duration', {int_or_none}),
+                'thumbnail': ('thumbnail', ('original_url', 'url'), {url_or_none}),
+            }), get_all=False)
+        }
+
+    def _extract_entries(self, video_ids, metainfo, list_id):
+        entries = [{**metainfo, **self._extract_video(id)} for id in set(video_ids)]
+        return self.playlist_result(entries, **metainfo, id=list_id)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        query = {self._INFO_QUERY_KEY: video_id}
+
+        data = traverse_obj(
+            self._get_api(self._INFO_API, video_id, query=query), self._DATA_PATH)
+
+        video_ids = traverse_obj(data, self._ID_PATH)
+        metainfo = traverse_obj(data, self._META_PATH)
+        return self._extract_entries(video_ids, metainfo, video_id)
+
+
+class TapTapIntlBase(TapTapBaseIE):
+    _X_UA = 'V=1&PN=WebAppIntl2&LANG=zh_TW&VN_CODE=115&VN=0.1.0&LOC=CN&PLT=PC&DS=Android&UID={uuid}&CURR=&DT=PC&OS=Windows&OSV=NT%208.0.0'
+    _VIDEO_API = 'https://www.taptap.io/webapiv2/video-resource/v1/multi-get'
+
+
+class TapTapMomentIE(TapTapBaseIE):
+    _VALID_URL = r'https?://www\.taptap\.cn/moment/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.cn/webapiv2/moment/v3/detail'
+    _ID_PATH = ('moment', 'topic', (('videos', ...), 'pin_video'), 'video_id')
+    _META_PATH = ('moment', {
+        'timestamp': ('created_time', {int_or_none}),
+        'uploader': ('author', 'user', 'name', {str}),
```

Comment:
Some additional metadata
```suggestion
        'modified_timestamp': ('edited_time', {int_or_none}),
        'uploader': ('author', 'user', 'name', {str}),
        'uploader_id': ('author', 'user', 'id', {int}, {str_or_none}),
```